openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -38,15 +41,44 @@ def _field_values(fields, xblock):
     return result
 
 
+def _has_assignment_blocks(item):
+    """
+    Check if a given block contains children that are assignments.
+    Assignments have graded, has_score and nonzero weight attributes.
+    """
+    def leaf_filter(block):
+        # Filter for traversals to find leaf blocks
+        return (
+            block.location.block_type not in ('chapter', 'sequential', 'vertical') and
+            len(block.get_children()) == 0
+        )
+
+    def get_children(parent):
+        # This function is used to get the children of a block in the traversal below
+        if parent.has_children:
+            return parent.get_children()
+        else:
+            return []
```

Comment:
That logic isn't just built into `get_children`?  :(

Reply:
I copied this function and the leaf_filter from another spot, but I think the idea is that get_children is defined on an object but we're passing this as a standalone function to the traversal